src/sshfs_mount/sshfs_mount.cpp

```diff
@@ -81,8 +105,19 @@ auto make_sftp_server(mp::SSHSession&& session, const std::string& source, const
              fmt::format("{}:{} {}(source = {}, target = {}, …): ", __FILE__, __LINE__, __FUNCTION__, source, target));
 
     check_sshfs_exists(session);
-    make_target_dir(session, target);
-    set_owner_for(session, target);
+
+    std::vector<std::string> splitting;
+    split_path(target, splitting);
+    bool needs_create = false;
+    for (std::string partial : splitting)
+    {
+        if (needs_create || !std::filesystem::exists(partial))
+        {
+            needs_create = true;
+            make_target_dir(session, partial);
+            set_owner_for(session, partial);
+        }
```

Comment:
We could optimize this by still running `mkdir --parents`, but only with the "remainder" of the path, and then `chown --recursive`, provided both processes' working dir was the subpath that does exist?

We'd then probably have to add an explicit check for file or directory, as changing `CWD` would fail if it was a file.

Reply:
If we do that, we'd have to check anyway ownership of each subpath, since we are not changing ownership of existing directories. Am I missing something?

Reply:
If we just do the equivalent of `cd /existing/paths; mkdir --parents remainder/to/be/created; chown … --recursive remainder/to/be/created`, we're good. So rather than `std::filesystem::exists`, use `std::filesystem::is_directory`.

It may also make sense to iterate in reverse.

Reply:
Done, implemented as Saviq and ChrisTownsend suggested.